lib/home_page.dart

```diff
@@ -32,31 +37,51 @@ class _HomePageState extends State<HomePage> {
     super.dispose();
   }
 
-  Widget buildList(List<Pokemon> data) {
-    return GridView.builder(
-      itemCount: data.length,
-      gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
-        crossAxisCount: NumericConstants.crossAxisCount,
-      ),
-      itemBuilder: (BuildContext context, int index) {
-        return OpenContainer(
-          openColor: Colors.black,
-          transitionType: ContainerTransitionType.fadeThrough,
-          transitionDuration: Duration(seconds: 1),
-          openBuilder: (context, _) => DetailScreen(
-            pokemon: data[index],
-          ),
-          closedBuilder: (context, openContainer) => InkWell(
-            onTap: () {
-              openContainer();
-            },
-            child: GridTilePokemons(
-              pokemon: data[index],
-            ),
-          ),
-        );
-      },
-    );
+  Widget buildList(PokemonResponse data) {
+    return PageView.builder(
+        physics: BouncingScrollPhysics(),
+        onPageChanged: (index) {
+          if (currentPage < _controller.page!) {
+            if (data.next != null) {
+              _pokemonBloc.fetchPokemonPage(
+                data.next!,
+              );
+            }
+          } else {
+            if (data.previous != null) {
+              _pokemonBloc.fetchPokemonPage(data.previous!);
+            }
+          }
```

Comment:
I think that this logic could be inside _pokemonBloc, because the bloc has the necessary information to do this. In home_page you only need to call "pokemonBloc.someMethod()" and said "give me next page" and inside the bloc he provides you a page

Reply:
Fixed